allennlp/evaluation/evaluator.py

```diff
@@ -0,0 +1,231 @@
+            Optional path to write the predictions to. If passed the
+            postprocessor will be called and its output will be written as lines.
+
+        # Returns
+
+        metrics: `Dict[str, Any]`
+            The metrics from evaluating the file.
+        """
+        raise NotImplementedError("__call__")
+
+
+@Evaluator.register("simple")
+class SimpleEvaluator(Evaluator):
+    """
+    Simple evaluator implementation. Uses the vanilla evaluation code.
+
+    # Parameters
+
+    batch_postprocessor: `Postprocessor`, optional (default=`SimplePostprocessor`)
+        The postprocessor to use for turning both the batches and the outputs
+        of the model into human readable data.
+
+    cuda_device : `Union[int, torch.device]`, optional (default=`-1`)
+        The cuda device to use for this evaluation.  The model is assumed to
+         already be using this device; this parameter is only used for moving
+         the input data to the correct device.
+    """
+
+    def __init__(
+        self,
+        batch_postprocessor: Optional[Postprocessor] = None,
+        cuda_device: Union[int, torch.device] = -1,
+    ):
+        super(SimpleEvaluator, self).__init__(batch_postprocessor, cuda_device)
+
+    def __call__(
+        self,
+        model: Model,
+        data_loader: DataLoader,
+        batch_weight_key: str = None,
+        output_file: Union[str, PathLike] = None,
+        predictions_output_file: Union[str, PathLike] = None,
+    ):
+        """
+        Evaluate a single data source.
+
+        # Parameters
+
+        model : `Model`
+            The model to evaluate
+        data_loader : `DataLoader`
+            The `DataLoader` that will iterate over the evaluation data (data loaders already contain
+            their data).
+        batch_weight_key : `str`, optional (default=`None`)
+            If given, this is a key in the output dictionary for each batch that specifies how to weight
+            the loss for that batch.  If this is not given, we use a weight of 1 for every batch.
+        metrics_output_file : `Union[str, PathLike]`, optional (default=`None`)
+            Optional path to write the final metrics to.
+        predictions_output_file : `Union[str, PathLike]`, optional (default=`None`)
+            Optional path to write the predictions to.
+
+        # Returns
+
+        metrics: `Dict[str, Any]`
+            The metrics from evaluating the file.
+        """
+        check_for_gpu(self.cuda_device)
+        data_loader.set_target_device(int_to_device(self.cuda_device))
+        output_file = Path(output_file) if output_file is not None else None
+        if predictions_output_file is not None:
+            predictions_file = Path(predictions_output_file).open("w", encoding="utf-8")
+        else:
+            predictions_file = None  # type: ignore
+
+        model_postprocess_function = getattr(model, "make_output_human_readable", None)
```

Comment:
Do we want the ability to pass in a function explicitly? I can imagine a case where people are using an older model to make some predictions, but want to write their own `postprocess_function`.

Reply:
I added the ability to pass a string for the function name from the model for postprocessing, it might be better to define a postprocessor registrable (I renamed the old postprocessor registrable to `serializer`).